Pokecon/Program.cs

```diff
@@ -85,22 +85,122 @@ static ResponseEnvelop api_req(string api_endpoint, string access_token, Request
             }
         }
 
-        static string get_api_endpoint(string access_token)
+        static string get_api_endpoint(string access_token, string provider)
         {
             var ret = api_req(API_URL, access_token, new[] {
                 new RequestEnvelop.Types.Requests { Type = 2 },
                 new RequestEnvelop.Types.Requests { Type = 126 },
                 new RequestEnvelop.Types.Requests { Type = 4 },
                 new RequestEnvelop.Types.Requests { Type = 129 },
                 new RequestEnvelop.Types.Requests { Type = 5, Message = new RequestEnvelop.Types.Unknown3 { Unknown4 = "4a2e9bc330dae60e7b74fc85b98868ab4700802e" } },
-            });
+            }, provider);
             try { return "https://" + ret.ApiUrl + "/rpc"; }
-            catch { return null; }
+            catch (Exception e){
+                return null;
+            }
+        }
+
+        static ResponseEnvelop get_profile(string api_endpoint, string access_token, string provider)
+        {
+            return api_req(api_endpoint, access_token, new[] { new RequestEnvelop.Types.Requests { Type = 2 } }, provider);
         }
 
-        static ResponseEnvelop get_profile(string api_endpoint, string access_token)
+        static string login_google(string username, string password)

```

Comment:
Some of the things you are doing in this method could be extracted into their own methods for easier readability and maintainability.